### ISSUE TITLE
[docs] Removing dead code instead of commenting it out

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#7422]
In https://github.com/organicmaps/organicmaps/pull/7333#issuecomment-1942472700 @biodranik is requesting the PR's author to comment out some old dead code instead of removing it:

> Nobody will look for some older code in git history without knowing that some code existed before. Commented code is very _explicit_ and acts as a TODO comment.

This is definitely something and controversial. In my opinion, the dead code shall be removed, not question asked. The code is dead when it is not used by the app and there is no way to run and test it. This efficiently means that there is no way to maintain it. 

My proposal is to follow the best industry practices and remove the dead code instead of commenting it out. If you need the somebody in the future, it can be always restored it from git.

Why objections?

CC: @organicmaps/contributors

**Comment by biodranik** [comment]
There are some features of OM that were commented, and will be highly likely restored again. If @vng or me are leaving commented code, then it means we plan to reuse it later.

Leaving commented code without any plans for future reuse (or without considering it helpful to restore/implement missing/new feature) is prohibited.

**Comment by pastk** [comment]
I agree with @biodranik that there might be special/rare cases when its beneficial to just comment-out the code. Its a case-by-case thing, but usually I'd assume that such code snippets are small, are deemed highly likely to be re-used and a comment is added to clarify the reason it wasn't just deleted.
